geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZAddIntegrationTest.java

```diff
@@ -0,0 +1,239 @@
+
+    long added = jedis.zadd(SORTED_SET_KEY, initMap);
+    assertThat(added).isEqualTo(INITIAL_MEMBER_COUNT);
+    // TODO: use ZCARD to confirm set size once command is implemented
+
+    for (String member : initMap.keySet()) {
+      Double score = initMap.get(member);
+      assertThat(jedis.zscore(SORTED_SET_KEY, member)).isEqualTo(score);
+    }
+
+    Map<String, Double> updateMap = makeMemberScoreMap("member_", 2 * INITIAL_MEMBER_COUNT, 10);
+
+    ZAddParams zAddParams = new ZAddParams();
+    zAddParams.nx();
+    added = jedis.zadd(SORTED_SET_KEY, updateMap, zAddParams);
+    assertThat(added).isEqualTo(INITIAL_MEMBER_COUNT);
+    // TODO: use ZCARD to confirm set size once command is implemented
+
+    for (String member : updateMap.keySet()) {
+      Double score;
+      if (initMap.get(member) != null) {
+        score = initMap.get(member);
+      } else {
+        score = updateMap.get(member);
+      }
+      assertThat(jedis.zscore(SORTED_SET_KEY, member)).isEqualTo(score);
+    }
+  }
+
+  @Test
+  public void zaddDoesNotAddNewMembers_whenXXSpecified() {
+    Map<String, Double> initMap = makeMemberScoreMap("member_", INITIAL_MEMBER_COUNT, 0);
+
+    long added = jedis.zadd(SORTED_SET_KEY, initMap);
+    assertThat(added).isEqualTo(INITIAL_MEMBER_COUNT);
+    // TODO: use ZCARD to confirm set size once command is implemented
+
+    for (String member : initMap.keySet()) {
+      Double score = initMap.get(member);
+      assertThat(jedis.zscore(SORTED_SET_KEY, member)).isEqualTo(score);
+    }
+
+    Map<String, Double> updateMap = makeMemberScoreMap("member_", 2 * INITIAL_MEMBER_COUNT, 10);
+
+    ZAddParams zAddParams = new ZAddParams();
+    zAddParams.xx();
+    added = jedis.zadd(SORTED_SET_KEY, updateMap, zAddParams);
+    assertThat(added).isEqualTo(0);
+    // TODO: use ZCARD to confirm set size once command is implemented
+
+    for (String member : updateMap.keySet()) {
+      Double score;
+      if (initMap.get(member) != null) {
+        score = updateMap.get(member);
+      } else {
+        score = null;
+      }
+      assertThat(jedis.zscore(SORTED_SET_KEY, member)).isEqualTo(score);
+    }
+  }
+
+  @Test
+  public void zaddDoesNotUpdateMember_whenNXSpecified() {
+    String key = "ss_key";
+
+    Long res = jedis.zadd(key, 1.0, "mamba");
+    assertThat(res).isEqualTo(1);
+
+    ZAddParams zAddParams = new ZAddParams();
+    zAddParams.nx();
+    res = jedis.zadd(key, 2.0, "mamba", zAddParams);
+    assertThat(res).isEqualTo(0);
+    assertThat(jedis.zscore(key, "mamba")).isEqualTo(1.0);
+  }
+
+  @Test
+  public void zaddDoesNotAddMember_whenXXSpecified() {
+    String key = "ss_key";
+
+    ZAddParams zAddParams = new ZAddParams();
+    zAddParams.xx();
+    Long res = jedis.zadd(key, 1.0, "mamba", zAddParams);
+    assertThat(res).isEqualTo(0);
+    assertThat(jedis.zscore(key, "mamba")).isEqualTo(null);
+    assertThat(jedis.exists(key)).isFalse();
+  }
```

Comment:
I'm not sure I understand how these tests are different from the ones above which do the same thing, but on multiple members.

Reply:
Mostly they're remnants of TDD. First a single case, then multiples. They do offer the slight benefit of using different Jedis ZADD calls (one that sets a single member, the other that takes a map of multiple members. Of course, we're not actually testing Jedis itself, so that benefit is pretty marginal. I'm cool with ditching the singletons.

Reply:
Ah, I see, that makes sense. I think it would be best to remove those tests though, since we now aren't getting any real value from them.

Reply:
Spiked.